pkg/git/git.go

```diff
@@ -8,17 +8,21 @@ import (
 	"strings"
 
 	"github.com/go-logr/logr"
-	"github.com/weaveworks/profiles/api/v1alpha1"
 	"k8s.io/apimachinery/pkg/util/yaml"
+
+	"github.com/weaveworks/profiles/api/v1alpha1"
 )
 
+// HTTPClient defines an http client which then can be used to test the
+// handler code.
 //go:generate counterfeiter -o fakes/fake_http_client.go . HTTPClient
 type HTTPClient interface {
 	Get(string) (*http.Response, error)
 }
 
 var httpClient HTTPClient = http.DefaultClient
 
+// GetProfileDefinition returns a definition based on a url and a branch.
 func GetProfileDefinition(repoURL, branch string, log logr.Logger) (v1alpha1.ProfileDefinition, error) {
```

Comment:
How about we change this to return a `*v1alpha1.ProfileDefinition` ?

This simplifies the return types, and also clarifies the ownership of the value?

Reply:
While I usually agree, here, I would like to avoid refactoring, because this PR only deals with the Linter.

That being said, I don't see why this should share up the stack? What would be the benefit if having this as a pointer? The related calls elsewhere, do use a value and then pass that value to other functions which do not require a pointer.

Could you elaborate your view a bit further please?

Reply:
Further, if this would be a pointer, definition in `Profile` would have to be a pointer and would thus require nil checks all over the place for valid values and property access.